src/main/java/com/atlassian/db/replica/internal/ReplicaStatement.java

```diff
@@ -39,6 +30,408 @@
     private final DatabaseCall databaseCall;
     final String methodBracketStart = Pattern.quote("(");
     private boolean isWriteOperation = true;
+    private final Set<String> readOnlyFunctions;
+    private static final Set<String> standardReadOnlyFunctions = Stream
```

Comment:
Ok, this guy is big
We can extract an `SqlFunctions` class, which could hold the list and the query parsing logic. Should be easier to unit-test too

Reply:
I was waiting for this comment 😉 . But seriously I'm testing it with the product, and I postponed refactoring.